packages/common/src/util/http.js

```diff
@@ -243,9 +243,10 @@ function encodeRequestBody(body) {
 }
 
 async function readResponseBody(response, parseAs) {
+  const contentLength = response.body?.readableLength ?? 0;
```

Comment:
why checking `contentLength` on response.body ? does this mean for contentType we should use `response.body` as well ?